tests/Spec.hs

```diff
@@ -1384,6 +1384,291 @@ main = hspec $ do
+          ]
+        results = defaultResults {resultIntegrate = [Right (Sha "1b2"), Right (Sha "2b2"), Right (Sha "3b2"), Right (Sha "4b2")]}
+        actions = snd $ runActionCustom results $ handleEventsTest events state
+      actions `shouldBe`
+        [ AIsReviewer (Username "deckard")
+        , ALeaveComment (PullRequestId 1) "<!-- Hoff: ignore -->\nPull request approved for merge by @deckard, rebasing now."
+        , ATryIntegrate "Merge #1: Improvements...\n\n\
+                        \Approved-by: deckard\n\
+                        \Auto-deploy: false\n"
+                        (PullRequestId 1, Branch "refs/pull/1/head",Sha "ab1")
+                        []
+                        False
+        , ALeaveComment (PullRequestId 1) "<!-- Hoff: ignore -->\nRebased as 1b2, waiting for CI …"
+        , ALeaveComment (PullRequestId 1) "<!-- Hoff: ignore -->\n[CI job :yellow_circle:](example.com/1b2) started."
+        , AIsReviewer (Username "deckard")
+        , ALeaveComment (PullRequestId 2) "<!-- Hoff: ignore -->\nPull request approved for merge by @deckard, waiting for rebase behind one pull request."
+        , ATryIntegrate "Merge #2: ... Of the ...\n\n\
+                        \Approved-by: deckard\n\
+                        \Auto-deploy: false\n"
+                        (PullRequestId 2,Branch "refs/pull/2/head",Sha "ab2")
+                        [PullRequestId 1]
+                        False
+        , ALeaveComment (PullRequestId 2) "<!-- Hoff: ignore -->\nSpeculatively rebased as 2b2 behind 1 other PR, waiting for CI …"
+        , ALeaveComment (PullRequestId 2) "<!-- Hoff: ignore -->\n[CI job :yellow_circle:](example.com/2b2) started."
+        , ALeaveComment (PullRequestId 1) "<!-- Hoff: ignore -->\nThe [build failed :x:](example.com/1b2).\n\nIf this is the result of a flaky test, then tag me again with the `retry` command.  Otherwise, push a new commit and tag me again."
+        , ATryIntegrate "Merge #2: ... Of the ...\n\n\
+                        \Approved-by: deckard\n\
+                        \Auto-deploy: false\n"
+                        (PullRequestId 2,Branch "refs/pull/2/head",Sha "ab2")
+                        []
+                        False
+        , ALeaveComment (PullRequestId 2) "<!-- Hoff: ignore -->\nRebased as 3b2, waiting for CI …"
+        , ALeaveComment (PullRequestId 2) "<!-- Hoff: ignore -->\nPush to master detected, rebasing again."
+        , ATryIntegrate "Merge #2: ... Of the ...\n\n\
+                        \Approved-by: deckard\n\
+                        \Auto-deploy: false\n"
+                        (PullRequestId 2,Branch "refs/pull/2/head",Sha "ab2")
+                        []
+                        False
+        , ALeaveComment (PullRequestId 2) "<!-- Hoff: ignore -->\nRebased as 4b2, waiting for CI …"
+        ]
+
+    it "does not restart integration after a push to master from a previous merge in the merge train" $ do
```

Comment:
This testname fits better with the test case a few tests below. A better name for this would be something like: `does not restart integration of failed builds after a push to master`.